www/include/options/accessLists/resourcesACL/DB-Func.php

```diff
@@ -45,15 +45,20 @@ function testExistence($name = null)
     if (isset($form)) {
         $id = $form->getSubmitValue('lca_id');
     }
-    $query = "SELECT acl_res_name, acl_res_id FROM `acl_resources` WHERE acl_res_name = '" . $name . "'";
-    $dbResult = $pearDB->query($query);
-    $lca = $dbResult->fetch();
-    if ($dbResult->rowCount() >= 1 && $lca["acl_res_id"] == $id) {
-        return true;
-    } elseif ($dbResult->rowCount() >= 1 && $lca["acl_res_id"] != $id) {
-        return false;
+    $name = filter_var($name, FILTER_SANITIZE_STRING);
```

Comment:
Why use this method ?
Useless when using the prepare method of PDO,

Reply:
this filter is used to remove the html entities and not to prevent injection.

Without the filter_var, the testExistence method can't match an existing result in DB and allow the creation.